Sources/UniRedis/UniRedis.swift

```diff
@@ -53,27 +54,31 @@ public class UniRedis {
 	}
 
 	public init(_ url: String) throws {
-		guard let match = try url.match("^(?:redis(\\+sentinel)?://)?(?:([^@]+)@)?([^:/]+)(?::([0-9]+))?(?:/([0-9]+))?$"), match.count == 5 else {
+		guard let match = try url.match("^(?:redis(\\+sentinel)?://)?(?:([^:@]+):)?(?:([^@]+)@)?([^:/]+)(?::([0-9]+))?(?:/([0-9]+))?$"), match.count == 6 else {
 			throw UniRedisError.error(detail: "failed to parse redis url '\(url)'")
 		}
 		sentinel = (match[0] != nil) ? true:false
-		host = match[2]!
-		if let m = match[3] {
+		host = match[3]!
+		if let m = match[4] {
 			port = Int32(m)!
 		}
-		if let m = match[4] {
+		if let m = match[5] {
 			db = Int(m)!
 		}
-		if let m = match[1] {
+		if let m = match[2] {
 			password = m
 		}
+		if let m = match[1] {
+			username = m
+		}
```

Comment:
As commented above, let's handle user:pass as a single token and try to split it here, like:

```swift
if let m = match[1] {
     if let auth = try m.match("^(.+):(.+)$") {
          username = auth[1]
          password = auth[2]
     } else {
          password = m
     }
}
```

Just proposal, not tested.